src/lua/stack/register_function.cr

```diff
@@ -0,0 +1,26 @@
+module Lua
+  module StackMixin::RegisterFunction
+    macro included
+      macro register_function(stack, name, proc)
+        {% verbatim do %}
+        proc = ->(state : LibLua::State) {
+          {% for arg, index in proc.args %}
+            {{ arg.name }} = LibLua.tonumberx(state, {{ index + 1 }}, nil)
+          {% end %}
+
+          # push result to stack
+          LibLua.pushnumber(state, {{ proc.body }})
```

Comment:
I'm assuming if one of the `proc.args` is a `String`, then we would call `LibLua.pushstring` here? But that expected a `Pointer(UInt8)` to be passed in... But also, what happens if the first arg is a String, and the second is a number? Does the value of each arg need to be pushed separately?